pkg/operator/deploy/deploy.go

```diff
@@ -432,6 +436,34 @@ func (o *operator) RenewMDSDCertificate(ctx context.Context) error {
 	return nil
 }
 
+func (o *operator) EnsureUpgradeAnnotations(ctx context.Context) error {
+	if o.oc.Properties.PlatformWorkloadIdentityProfile == nil ||
+		o.oc.Properties.ServicePrincipalProfile != nil {
```

Comment:
Why both conditions are required here? Why can't we rely on either of the 2 conditions?

Reply:
I was thinking belt and braces, but which one do you think makes more sense to rely on?

Reply:
The more I think about this, the more I think we should have a utility function that we can call everywhere in the codebase that runs this check so we don't have to duplicate the logic everywhere. I think we could probably handle this here or in a separate PR.

Reply:
Yes, I agree, and better to have a standardized way(util function) to check this part.

Reply:
I'll card up the utility function and have Sanjana or Alex handle :)

Reply:
I would also go with `||`, we don't want to perform any MIWI operations until all the conditions we think are required to be termed it as a MIWI cluster are `True`.